Modules/posixmodule.c

```diff
@@ -5490,12 +5490,43 @@ os__path_normpath_impl(PyObject *module, PyObject *path)
         return NULL;
     }
     Py_ssize_t norm_len;
-    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, &norm_len);
+    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, 0, &norm_len);
     PyObject *result = PyUnicode_FromWideChar(norm_path, norm_len);
     PyMem_Free(buffer);
     return result;
 }
 
+/*[clinic input]
+os._path_abspath
+
+    path: object
```

Comment:
Could you use the following?
```suggestion
    path: unicode
```
Than the unicode check is performed by the argument clinic generated code. (note: I am not sure about the error message that might be different)

Reply:
Yes, that's worth trying.

Reply:
Yeah, it's better to let argument clinic handle that.

Reply:
I'm not sure if it's still worth it when the function has a single argument.

Reply:
> I'm not sure if it's still worth it when the function has a single argument.

Why?

Reply:
Because the function is currently called directly, and isn't called by a wrapper function:

https://github.com/python/cpython/blob/fff561e70f14ce57aa3c2b8d2c3df7325e2ea0bd/Modules/clinic/posixmodule.c.h#L2306-L2317

Reply:
I don't understand; why is it not worth it to let Argument Clinic handle the type checking of the `path` argument; why would you want to spell that out in C code?

Reply:
Nevermind, it's probably just premature optimisation.